### DISCUSSION TITLE
Socket.io and WebSocket proxy

### DISCUSSION THREAD

**Discussion by freez10** [TypedProject/ts-express-decorators#2136] [General]
Hi there!

I have WebSocket proxy configured like so
``` 
  $beforeRoutesInit() { // Doing same in $afterRoutesInit() { makes no difference
    this.app
      .use(
        '/subscriptions',
        createProxyMiddleware({
          target: 'ws://localhost:8080',
          changeOrigin: true,
          ws: true,
        }),
      );
  }
```

Now if I try to add any SocketIO service, proxy stops working.

Is it possible having WebSocket proxy and SocketIO services at the same time somehow?

**Comment by Romakita** [comment]
@freez10 I suggest you to create reproducible example. What is your error message? Have you an original example with express and socket io ?

Probably the proxy fail because the socket io server catch all ws connection.

**Reply by freez10** [reply]
Thanks for your reply @Romakita 

It turned out one have to pass a context to the proxy middleware.
This way it works as expected.

```
    this.app
      .use(
        createProxyMiddleware('/subscriptions', {
          target: 'ws://localhost:8080',
          changeOrigin: true,
          ws: true,
        }),
      )
```
